calculator_test.go

```diff
@@ -24,31 +24,43 @@ func TestSubtract(t *testing.T) {
 }
 
 type TestCase struct {
-	A    float64
-	B    float64
+	a    float64
+	b    float64
+	f    func(float64, float64) float64
 	want float64
+	name string
 }
 
-func TestMultiply(t *testing.T) {
+func TestCalculator(t *testing.T) {
 	t.Parallel()
 	testCases := []TestCase{
 		{
-			A:    2,
-			B:    2,
+			a:    2,
+			b:    2,
+			f:    calculator.Add,
 			want: 4,
+			name: "Add",
 		},
 		{
-			A:    -3,
-			B:    3,
-			want: -9,
+			a:    3,
+			b:    5,
+			f:    calculator.Subtract,
+			want: -2,
+			name: "Subtract",
+		},
+		{
+			a:    4,
+			b:    20,
+			f:    calculator.Multiply,
+			want: 80,
+			name: "Multiply",
 		},
 	}
 
 	for _, tc := range testCases {
-		got := calculator.Multiply(tc.A, tc.B)
+		got := tc.f(tc.a, tc.b)
 		if tc.want != got {
-			t.Errorf("Multiply(%f, %f): want %f, got %f", 2.0, 2.0, tc.want, got)
+			t.Errorf("%s: want %f, got %f", tc.name, tc.want, got)
```

Comment:
We can use `t.Run` here to avoid having to explicitly print the test case name:

```go
for ... {
	t.Run(tc.name, func(t *testing.T) {
		... // test code goes here
```